p24/src/refs.rs

```diff
@@ -0,0 +1,127 @@
+/// Mutable reference to first element if flag is false, second if true
+pub fn f1(tuple: &mut (u32, u32, bool)) -> &mut u32 {
+    if tuple.2 { &mut tuple.1 } else { &mut tuple.0 }
+}
+
+/// Returns mutable reference to n-th element in slice
+pub fn f2(slice: &mut [u32], n: usize) -> &mut u32 {
+    &mut slice[n]
+}
+
+/// Returns mutable reference to n-th element from end
+/// ///
+/// # Examples
+/// ```
+/// use p24::refs::f3;
+/// let mut arr = [1, 2, 3, 4, 5];
+/// *f3(&mut arr, 1) = 9;
+/// assert_eq!(arr, [1, 2, 3, 9, 5]);
+/// ```
+///
+pub fn f3(slice: &mut [u32], n: usize) -> &mut u32 {
+    let idx = slice.len() - 1 - n;
```

Comment:
Note that this may underflow, which does not panic when compiled in release mode. In the case if `n` is close to `usize::MAX` this may result in a valid `idx`. So it's worth to add a check that `n` is less than `slice.len()`.